src/Sarif.Multitool/Rules/RuleResources.resx

```diff
@@ -210,4 +210,10 @@
   <data name="SARIF1019_RuleIdMustBePresentAndConsistent" xml:space="preserve">
     <value>In every result, at least one of the properties result.ruleId and result.rule.id must be present. If both are present, they must be equal.</value>
   </data>
+  <data name="SARIF1020_SchemaMustBePresentAndConsistent" xml:space="preserve">
+    <value>The schema must be presented and consistent.</value>
+  </data>
+  <data name="SARIF1020_UseNewSchemaVersion" xml:space="preserve">
+    <value>The schema version used is old.</value>
+  </data>
```

Comment:
I will probably need some help with the messages!

Reply:
No problem. There are two messages:

1. When the `$schema` property is absent:

"The SARIF log file does not contain a $schema property. Add a $schema property that refers to the final version of the SARIF 2.1.0 schema. This enables IDEs to provide Intellisense for SARIF log files."

2. When the `$schema` property is present but does not end with `/sarif-2.1.0.json` or `/sarif-2.1.0-rtm.5.json`:

"The $schema property does not refer to the final version of the SARIF 2.1.0 schema. If you are using an earlier version of the SARIF format, consider upgrading your analysis tool to produce the final version. If this file does in fact conform to the final version of the schema, upgrade the tool to populate the `$schema` property with a URL that refers to the final version of the schema."

---
In reply to: [433577636](https://github.com/microsoft/sarif-sdk/pull/1892#discussion_r433577636) [](ancestors = 433577636)